processing/src/main/java/org/apache/druid/segment/filter/AndFilter.java

```diff
@@ -257,12 +264,12 @@ public boolean equals(Object o)
       return false;
     }
     AndFilter andFilter = (AndFilter) o;
-    return Objects.equals(getFilters(), andFilter.getFilters());
+    return Objects.equals(hashCode(), andFilter.hashCode());
```

Comment:
equals should not compare hashCodes. It's possible that different and filters hash to the same hashCode.

Similar comments in the other filters